gobblin-metastore/src/main/java/org/apache/gobblin/metastore/metadata/StateStoreEntryMetadata.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.metastore.metadata;
+
+import java.io.IOException;
+
+import org.apache.gobblin.configuration.State;
+import org.apache.gobblin.metastore.StateStore;
+
+
+import lombok.Data;
+
+
+/**
+ * Contains metadata about an entry in a {@link StateStore}.
+ * @param <T> type of {@link State} that can be read from this entry.
+ */
+@Data
+public abstract class StateStoreEntryMetadata<T extends State> {
```

Comment:
Is it more like a `StateStoreEntryManager`, that contains the reference to the StateStore that this entry exists, and access the state or delete it from statestore if necessary ? 

Names with a verb in it looks closer to the way it is functioning.